### PR TITLE
fix: Deep zoom issues

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3186] 

### Description

https://user-images.githubusercontent.com/11945712/206481214-8c59036b-88b8-4ea1-9023-186d730cece1.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Fix deep zoom android issues - mounir

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3186]: https://artsyproduct.atlassian.net/browse/CX-3186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/ImageCarousel/FullScreen/ImageZoomViewAndroid.tsx [hunk 1]
```diff
@@ -67,13 +71,21 @@ export const ImageZoomViewAndroid: React.FC<ImageZoomViewAndroidProps> = ({ imag
...
+              setIsLoading(true)
+            })
           }}
           onLoadEnd={() => {
-            setIsLoading(false)
+            requestAnimationFrame(() => {
+              LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+              setOpacity(1)
+              setIsLoading(false)
+            })
```
Comment (MounirDhahri) [inline src/app/Scenes/Artwork/Components/ImageCarousel/FullScreen/ImageZoomViewAndroid.tsx:88]: using opacity with the default `easeInEaseOut` animation to have a better UX

File: src/app/Scenes/MyProfile/DevicePrefsModel.ts [hunk 2]
```diff
@@ -3,16 +3,23 @@ import { action, Action, computed, Computed } from "easy-peasy"
 import { Appearance } from "react-native"
 
 export interface DevicePrefsModel {
+  sessionState: {
```
Comment (MounirDhahri) [inline src/app/Scenes/MyProfile/DevicePrefsModel.ts:6]: no migration is needed here because we are using `sessionState`

Comment (dariakoko) [reply src/app/Scenes/MyProfile/DevicePrefsModel.ts:6]: niiice

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarouselEmbedded.tsx [hunk 3]
```diff
@@ -73,6 +75,11 @@ export const ImageCarouselEmbedded: React.FC<ImageCarouselEmbeddedProps> = ({
       (Platform.OS === "android" && !!enableAndroidImagesGallery && !disableFullScreen)
     ) {
       dispatch({ type: "TAPPED_TO_GO_FULL_SCREEN" })
+      // This is here to avoid a bug where the modal would show up with the dev menu at the same
+      // time while in dev. This won't happen in prod for users though
+      if (Platform.OS === "android") {
+        setIsDeepZoomModalVisible(true)
+      }
```
Comment (dariakoko) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarouselEmbedded.tsx:82]: Niiice

**Review by dariakoko — APPROVED** [review]
Thanks Mounir!
So simple and so hmmm not 😅

**Comment by dariakoko** [comment]
Let's resolve conflicts and merge

**Pull request merged**